### PR TITLE
PlotLayer add axes titles

### PR DESCRIPTION
**Pull request by Firenze11**
Allow passing in X, Y, Z axes title strings, displaying them along the axes.
image

### CODE REVIEW

File: examples/plot/app.js [hunk 1]
```diff
@@ -46,17 +46,31 @@ class Root extends Component {
...
         {...viewport}
         onChangeViewport={this._onChangeViewport.bind(this)} >
-        <DeckGLOverlay viewport={viewport}
+        <DeckGLOverlay
+          viewport={viewport}
           equation={equation}
           resolution={200}
-          showAxis={true} />
+          showAxis={true}
+          onHover={this._onHover.bind(this)} />
```
Comment (ibgreen) [inline examples/plot/app.js]: Better to pre-bind this in constructor as `Function.bind()` generates a new function every render and can thus defeat `shouldComponentUpdate`.

File: examples/plot/package.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
     "build": "NODE_ENV=production webpack --env.prod=true"
   },
   "dependencies": {
+    "d3-scale": "^1.0.6",
```
Comment (ibgreen) [inline examples/plot/package.json]: Is this used in the PR? Or was it a missing dependency?

Comment (shaojingli) [inline examples/plot/package.json]: same question... is d3-scale used for scaling each axis?

File: examples/plot/plot-layer/axes-layer.js [hunk 3]
```diff
@@ -254,23 +256,42 @@ export default class AxesLayer extends Layer {
...
+
   // updates the instancePositions and instanceNormals attributes
   calculateTicks(bounds, ticksCount) {
     const xTicks = getTicks(bounds[0], ticksCount);
     const yTicks = getTicks(bounds[1], ticksCount);
     const zTicks = getTicks(bounds[2], ticksCount);
 
-    return [xTicks, yTicks, zTicks];
+    return [
+      [...xTicks, (bounds[0][0] + bounds[0][1]) / 2],
```
Comment (shaojingli) [inline examples/plot/plot-layer/axes-layer.js]: Does this PR handles the case that different axis has different ranges? You once mentioned that you'd like to scale each axis differently.

**Comment by shaojingli** [comment]
Also.. github says you have not signed CLA

File: examples/plot/package.json (hunk 2 above)
Comment (Firenze11) [inline examples/plot/package.json]: This was a missing dependency before the PR was added. Yes it is used to scale the axes.

File: examples/plot/plot-layer/axes-layer.js (hunk 3 above)
Comment (Firenze11) [inline examples/plot/plot-layer/axes-layer.js]: Not yet. The solution to axes scaling was implemented by @Pessimistress probably on some other branch.

File: examples/plot/app.js (hunk 1 above)
Comment (Firenze11) [inline examples/plot/app.js]: Just updated.

File: examples/plot/plot-layer/README.md [hunk 4]
```diff
@@ -95,3 +95,9 @@ Amount that grids should setback from the bounding box. Relative to the size of
 - Default: `[0, 0, 0, 255]`
 
 Color to draw the grids with, in `[r, g, b, a]`.
+
+##### `axesTitles` (Array, optional)
+
+- Default: `['x', 'z', 'y']`
+
+Strings to draw next to each axis as their titles, note that the second element is the axis that points upwards.
```
Comment (Firenze11) [inline examples/plot/plot-layer/README.md]: Updated README

**Pull request merged**